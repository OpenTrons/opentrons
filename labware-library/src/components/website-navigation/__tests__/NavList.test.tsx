// tests for main navbar link list
import * as React from 'react'
import { shallow, mount } from 'enzyme'

import { NavList } from '..'

describe('NavList', () => {
  it('component renders', () => {
    const tree = shallow(<NavList />)

    expect(tree).toMatchSnapshot()
  })

  it('component handles state changes', () => {
    const tree = shallow(<NavList />)

    // @ts-expect-error(IL, 2021-03): state is readonly, but also, this is a weird test??
    expect(tree.state().menu).toEqual(null)
    tree.setState({ menu: 'foo' })
    // @ts-expect-error(IL, 2021-03): state is readonly, but also, this is a weird test??
    expect(tree.state().menu).toEqual('foo')
  })

  it('component applies active class to nav links based on state', () => {
    const wrapper = mount(<NavList />)

    // when state.menu is null, all nav links are active
    expect(wrapper.find('.nav_link').every('.active')).toBe(true)
    wrapper.setState({ menu: 'About' })
    // setting menu to a valid value activates just the one link
    expect(wrapper.find('.active.nav_link')).toHaveLength(1)
  })
})
