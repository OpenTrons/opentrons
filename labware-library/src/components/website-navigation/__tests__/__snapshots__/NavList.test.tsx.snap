// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavList component renders 1`] = `
<ClickOutside
  onClickOutside={[Function]}
>
  <Component />
</ClickOutside>
`;
